tutorials/persistance-comparison-series/PlayerPrefs/player_prefs.md

```diff
@@ -0,0 +1,436 @@
+
+```cs
+private void Start()
+{
+    // Check if the key exists. If not, we never saved the hit count before.
+    if (PlayerPrefs.HasKey(HitCountKeyUnmodified)) // 4
+    {
+        // Read the hit count from the PlayerPrefs.
+        hitCountUnmodified = PlayerPrefs.GetInt(HitCountKeyUnmodified); // 5
+    }
+    if (PlayerPrefs.HasKey(HitCountKeyShift)) // 4
+    {
+        // Read the hit count from the PlayerPrefs.
+        hitCountShift = PlayerPrefs.GetInt(HitCountKeyShift); // 5
+    }
+    if (PlayerPrefs.HasKey(HitCountKeyControl)) // 4
+    {
+        // Read the hit count from the PlayerPrefs.
+        hitCountControl = PlayerPrefs.GetInt(HitCountKeyControl); // 5
+    }
+}
+```
+
+As before, we first check if the key exists in the `PlayerPrefs` (4) and if so, we set the corresponding counter (5) to its value. This is fine for a simple example but here you can already see that saving more complex data will bring `PlayerPrefs` very soon to its limits if you do not want to write a lot of boilerplate code.
+
+Unity offers a detection for keyboard clicks and other input like a controller or the mouse via a class called [`Input`](https://docs.unity3d.com/ScriptReference/Input.html). Using `GetKey` we can check if a specific key was held down the moment we register a mouse click.
+
+The documentation tells us about one important fact though:
+
+> Note: Input flags are not reset until Update. You should make all the Input calls in the Update Loop.
+
+Therefore we also need to implement the [`Update()`](https://docs.unity3d.com/ScriptReference/MonoBehaviour.Update.html) function (6) where we check for the key and save it in the previously defined `keyPressed`.
+
+The keys can be addressed via their name as string but the type safe way to do this is to use the class `KeyCode` which defines every key necessary. For our case this would be `KeyCode.LeftShift` and `KeyCode.LeftControl`.
+
+Those checks use `Input.GetKey()` (7) and if one of the two was found, it will be saved as the `keyPressed` (8). If neither of them was pressed (9) we just reset `keyPressed` to the `default` (10) which we will use as a marker for an unmodified mouse click.
+
+```cs
+private void Update() // 6
+{
+    // Check if a key was pressed.
+    if (Input.GetKey(KeyCode.LeftShift)) // 7
+    {
+        // Set the LeftShift key.
+        keyPressed = KeyCode.LeftShift; // 8
+    }
+    else if (Input.GetKey(KeyCode.LeftControl)) // 7
+    {
+        // Set the LeftControl key.
+        keyPressed = KeyCode.LeftControl; // 8
+    }
+    else // 9
+    {
+        // In any other case reset to default and consider it unmodified.
+        keyPressed = default; // 10
+    }
+}
+```
+
+The same triplet can then also be found in the click detection:
+
+```cs
+private void OnMouseDown()
+{
+    // Check if a key was pressed.
+    if (keyPressed == KeyCode.LeftShift) // 11
```

Comment:
Maybe this will be more readable as a switch?

Reply:
Agree!
d03481e